### DISCUSSION TITLE
Insomnia doing Postman stuff?

### DISCUSSION THREAD

**Discussion by timozn** [Kong/insomnia#6565] [Feedback]
In the newest v2 API and the beta client 8.0.0, Collections will be removed unless you login.
So no more Collections in Insomnia unless you create an account and login.
Environments then are no longer linked to Collections, but globally available, as there is now only one collection "Scratch Pad".

This could all be fine, however [only allowing Plugins when logged in](https://github.com/Kong/insomnia/pull/6135/files#diff-8513427d39f0d04af41be761fdbbc8a44ec8ee7c2847f81117cfe0a00cbaa686R284) is making us having second thoughts about Insomnia.

Having to create an account, and therefore having the possibility to sync and  send data into the cloud, that could potentially contain credentials, is a huge red flag for our company, that made us move away from Postman.

Now seeing that Insomnia is planning to do the same thing makes me somehow sad.

Or maybe I am completely misinterpreting something here?

**Comment by subnetmarco** [comment]
Hi @timozn, a couple of clarifications:

1) We do support Scratch Pad, which is always local. Scratch Pad has some limitations, which can be overcome by creating an account, which brings me to the second point.
2) Unlike Postman, Insomnia is fully end-to-end encrypted (E2EE) on all subscription plans, including the Free plan. Which means that your data is always secure and nobody can look into it, not even us. Which is why storing the passphrase is a big deal with Insomnia, because losing it may cause loss of access to your data. Again, this is not just encryption, it is **end-to-end** encryption.
3) Furthermore, if you choose to synchronize your collections with Git Sync, your data never goes to our servers. At that point, we use the cloud to only manage your account (and invitations, collaboration, etc) but we never see your data.

Plenty of options to keep using Insomnia and still avoid sending us collections. And even if you did, they are safe with E2EE.

Hope this clarifies the situation, let me know if you have more questions.

**Reply by jsejcksn** [reply]
How do I access all of my existing data without signing in?

**Reply by subnetmarco** [reply]
You can export it before continuing with the cloud synchronization. Or you can export it after creating an account.

**Reply by jsejcksn** [reply]
As a longtime user of Insomnia without ever having created an account or signing in, it's quite surprising to open the app today after an update and find all of my data absent without any kind of notice…

> You can export it before continuing with the cloud synchronization. Or you can export it after creating an account.

Is this process clearly documented somewhere? Can you link me to that documentation or explain the steps I need to take? I want to access my data without creating an account or signing in. The only views that I seem to be able to access now are these:

<img width="1280" alt="Screenshot 2023-09-27 at 14 32 25" src="https://github.com/Kong/insomnia/assets/868251/a45e062b-8afc-4ec3-b7b2-3ba2a82cb369">

<img width="1280" alt="Screenshot 2023-09-27 at 14 32 33" src="https://github.com/Kong/insomnia/assets/868251/7127a58b-dcec-4e19-a653-0d75fdfeb0b6">

**Reply by subnetmarco** [reply]
You can downgrade to a previous version of Insomnia, by downloading it from here: https://github.com/Kong/insomnia/releases

**Reply by CleverCoder** [reply]
That's not a good answer. Yeah. I'm upset. Read this. https://en.wikipedia.org/wiki/Don%27t_be_evil

**Reply by subnetmarco** [reply]
We are shipping today a new version that simplifies the export https://github.com/Kong/insomnia/discussions/6590#discussioncomment-7137469

**Comment by penguinxr2** [comment]
This 8.0 update was truly awful. Causing data loss without a required login is a terribly dark patttern. Definitely going to find an alternative now.

**Reply by subnetmarco** [reply]
I want to assure you there was no data loss. You could export your data on the onboarding, you can downgrade your version of Insomnia (https://github.com/Kong/insomnia/releases) and still see it (you can even export it) or you can wait until EoW when we ship 8.1 that makes all of the above more clear.

**Comment by yjlcoder** [comment]
Today Insomnia on my machine automatically upgraded to 8.0.0, and I found the same issue. I removed the new version and re-downloaded the last version before 8.0.0. Luckily all of my collections are still there. I have disabled auto update in preferences and will start looking for a new alternative.

Insomnia was good while it's good. For a foreseeable future, I'll stay on this old version.

**Comment by jeronimoamazon** [comment]
This is a tremendous breach of trust for users who were using this in enterprise environments where the options for clients are very limited. The removal of offline collections and requiring logins for all this makes this tool a non-starter.

**Reply by subnetmarco** [reply]
You can downgrade to an older version and still be able to export your data.

**Reply by jeronimoamazon** [reply]
Are we able to re-import it and use the collections without logging in after upgrading back?

**Reply by subnetmarco** [reply]
You can re-import a collection in scratch pad and still be able to use it locally, yes.

**Reply by jeronimoamazon** [reply]
Ok cool. That worked. 

1. Downgrade to https://github.com/Kong/insomnia/releases/tag/core%402023.5.8
2. Export my collections
3. Upgrade again
4. Import each collection, make a folder, then drag each call into a folder

With this I can emulate what I had. That's kind of annoying but at least I can stay on the next version. It'd be easier if I could import into a folder or bulk drag, but maybe you can and I'm just an idiot.

**Comment by CleverCoder** [comment]
AWFUL. Not going to take the time to explain. I understand the need to monetize, but it's OPEN SOURCE. Figure it out. Now looking for a new alternative. Thankfully, in the meantime, the old version still works. (This was after screaming at my computer tonight because Stoplight Studio pulled similar garbage, requiring signing in and paying for things that were once free)....

**Comment by H3adcra5h** [comment]
This is the most horrible move you can make after Postman has switched to cloud compulsion. I often work in closed environments where I can't connect to the Internet. In these environments I cannot and I am not allowed to store data on third-party servers. This makes Insomnia absolutely unusable for me.

**Reply by subnetmarco** [reply]
Wouldn't the local Scratch Pad be the option you are looking for?

**Reply by H3adcra5h** [reply]
I did not test the newest version. But I do not have a connection to servers outside the secured environment and on the pictures I see I have to login. I cannot and I am not willig to use software that forcing me to use a cloud.
So R.I.P Insomnia.

**Reply by subnetmarco** [reply]
You don't need to login with the local Scratch Pad. It's fully offline.

**Reply by timozn** [reply]
How do I use Plugins without logging in?
Since there is no Auth-Inherit, I could implement something similar using plugins.

But Plugins now are only manageable when logged in.
Most likely because you put stuff like GitSync behind Login and/or Paywall and don't wan't users to simply use these plugins instead.
I can't see another reason for that, since plugins are running on my machine and not in the "cloud".

**Reply by subnetmarco** [reply]
Plugins are going to be re-enabled today or tomorrow in 8.1 in Scratch Pad, it was an oversight.

**Comment by subnetmarco** [comment]
I left an update here https://github.com/Kong/insomnia/discussions/6590#discussioncomment-7137469

**Comment by subnetmarco** [comment]
This discussion is a dup of https://github.com/Kong/insomnia/discussions/6590.

**Reply by Aeolun** [reply]
You locked the dup though.

While I do not think you’re going to get much useful discussion about anything, locking both threads makes you look like a tyrant. I also imagine that means the frustration will be let out in a different way, which might not be positive.

For what it’s worth, I don’t think anyone begrudges you the need to make money. 

What they do not like is the bait and switch. From one day to the other a release that was supposed to be a happy occasion lobotomized what was the v7 version and replaced it with a completely different app. There are people that can like your new app (a specific kind of enterprise market maybe?), but it’s not the same as the audience you had before.

Meanwhile that audience will be angry that the app that they –in some cases– fled to after Postman pulled the same stunt, destroys that trust by pulling the exact same one.

**Reply by subnetmarco** [reply]
The Insomnia 8.1 release was entirely driven by that conversation, but I don't think we were having a productive conversation anymore:

* Data stays with the user with Git Sync.
* Data stays local with Scratch Pad.
* Data can be exported.
* Data is end-to-end encrypted.
* Data is not automatically migrated to the cloud unless the user accepts to do so.
* Account information is stored in Auth0, which is ISO27001, SOC 2 Type II, ISO27018, HIPAA BAA, Gold CSA STAR, PCI and GDPR compliant.

It's the best compromise at the moment to both meet the data-concerned users in the middle, and also continue to build the product.

> locking both threads makes you look like a tyrant.

I have been a very active maintainer on GitHub on all channels, but I am not going to tolerate insults to the project or the team, that's where I draw the line.

**Reply by Aeolun** [reply]
> I am not going to tolerate insults to the project or the team

While I get you may have to do this in the situation you are in now, you should sort of have expected that? 

What’d you call your electricity company if they suddenly told you that from now on you can only use electricity while signed into their online service, but don’t worry, you can log out and use devices up to 200w while offline?

While you obviously shouldn’t necessarily listen to a majority of very vocal users, the complaint that v7 to v8 is a clear regression for some (mostly those with multiple local collections) is absolutely true. I’m really not sure how that stacks up to the entire userbase of Insomnia, it may ultimately be for the better.

Wouldn’t be the first time that I’ve seen the alienation of the entire early adopters base turn out to be wildly profitable because of a few now possible enterprise deals.

Personally, I’m not a fan of all my stuff being called a scratchpad for entirely irrational reasons, but it’ll likely make me switch to something else 🤷‍♂️.
